src/App.tsx

```diff
@@ -1,11 +1,55 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Filter } from './types/Filter';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState(false);
+  const [errorMessage] = useState('');
+
+  if (error) {
+    setTimeout(() => {
+      setError(false);
+    }, 3000);
+  }
+
+  const userId = 2;
+
+  if (!user) {
+    // userId = user.id;
+  }
+
+  getTodos(userId)
+    .then(setTodos)
+    .catch(() => setError(false));
```

Comment:
```suggestion
  getTodos(user?.id)
    .then(setTodos)
    .catch(() => setError(false));
```
useEffect

Reply:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code